flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java

```diff
@@ -574,7 +574,10 @@ public Object convert(Object dbzObj, Schema schema) throws Exception {
                     if (field == null) {
                         row.setField(i, null);
                     } else {
-                        Object fieldValue = struct.get(field);
+                        Object fieldValue =
+                                field.schema().isOptional()
+                                        ? struct.getWithoutDefault(fieldName)
+                                        : struct.get(field);
```

Comment:
Should we directly use `Object fieldValue = struct.getWithoutDefault(fieldName);`, the CDC process should never change the original data,  but only capture and transfer it. The assignment of Default value should happen in DB server side.

Reply:
I suddenly understood your means. The logic of my code looks stupid.haha....